port/zos390/omrsl.c

```diff
@@ -91,9 +101,73 @@ omrsl_open_shared_library(struct OMRPortLibrary *portLibrary, char *name, uintpt
 		handle = dllload(openName);
 	}
 
-	if (NULL == handle) {
+#if defined(J9ZOS39064)
+	/* Attempt to load target library as 31-bit. */
+	if (attempt31BitDLLLoad && (NULL == handle) && omr_cel4ro31_is_supported()) {
+		OMR_CEL4RO31_infoBlock *ro31InfoBlock = NULL;
+
+		/* Generate trace event for the error reason from preceding 64-bit dllload(). */
 		getDLError(portLibrary, errBuf, sizeof(errBuf));
 		Trc_PRT_sl_open_shared_library_Event2(errBuf);
+		isErrBufPopulated = TRUE;
+
+		DMESSAGE(("omrsl_open_shared_library: Attempt to load 31-bit DLL [%s]\n", openName))
+
+		/* Due to temporary LE limitation, we need to pass environment variables via _CEE_ENVFILE.
+		 * Since this could be the first path in establishing 31-bit secondary enclave,
+		 * and launcher may unset _CEE_ENVFILE in order for child processes to
+		 * pick up LIBPATH updates, we need to copy the _CEE3164_ENVFILE env var
+		 * to _CEE_ENVFILE for duration of this CEL4RO31 invocation.
+		 */
+		char *CEE_ENVFILE_envvar = getenv("_CEE_ENVFILE");
+		char *CEE3164_ENVFILE_envvar = getenv("_CEE3164_ENVFILE");
+		BOOLEAN initCEE_ENVFILE = ((NULL == CEE_ENVFILE_envvar) || ('\0' == CEE_ENVFILE_envvar[0])) && (NULL != CEE3164_ENVFILE_envvar);
+
+		if (initCEE_ENVFILE) {
+			char buf[1024];
+			snprintf(buf, sizeof(buf), "_CEE_ENVFILE=%s", CEE3164_ENVFILE_envvar);
+			putenv(buf);
+		}
+
+		ro31InfoBlock = omr_cel4ro31_init(OMR_CEL4RO31_FLAG_LOAD_DLL, openName, NULL, 0);
+		if (NULL != ro31InfoBlock) {
+			OMR_CEL4RO31_controlBlock *ro31ControlBlock = &(ro31InfoBlock->ro31ControlBlock);
+
+			omr_cel4ro31_call(ro31InfoBlock);
+
+			if (OMR_CEL4RO31_RETCODE_OK == ro31ControlBlock->retcode) {
+				/* Extract DLL handle from CEL4RO31 load result. */
+				handle = (void *)ro31ControlBlock->dllHandle;
+				/* High tag the handle to distingush 31-bit DLL. */
+				handle = (void *)(OMRPORT_SL_ZOS_31BIT_TARGET_HIGHTAG | (uintptr_t)handle);
+			} else {
+				/* Append the CEL4RO31 diagnostics to errBuf. */
+				char cel4ro_errBuf[128];
+				snprintf(cel4ro_errBuf,
```

Comment:
https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#function-parameter-list-does-not-fit-on-one-line

Reply:
Fixed under 813bcfe.